### ISSUE TITLE
Failed to join (setRedirect = false)

### ISSUE THREAD

**Issue by sazanof** [bigbluebutton/bigbluebutton-api-php#34]
Hello! I want to get right sessionToken to use it manualy. Code, I used:
```
$joinMeetingParams = new JoinMeetingParameters($m['meetingID'], $name, $pass);
if ($_POST['type'] == 'html5'){
    $joinMeetingParams->setJoinViaHtml5(true);
}
$joinMeetingParams->setRedirect(false);
$url = $bbb->getJoinMeetingURL($joinMeetingParams);
$joinMeetingResponse = $bbb->joinMeeting($joinMeetingParams);
var_dump($url,$joinMeetingResponse);
```
$url I got is 
**https://server/bigbluebutton/api/join?meetingID=356a192b7913b04c54574d18c28d46e6395428ab_8f9329121a2844579d346a7f5660372114b08ccf&fullName=Михаил&password=123456&redirect=false&joinViaHtml5=true&checksum=9bd0e52bb2d1cbce9f9ef508365ce80d361ac3b4**
$joinMeetingResponse I got is:
```
BigBlueButton\Responses\JoinMeetingResponse Object
(
    [rawXml:protected] => SimpleXMLElement Object
        (
            [returncode] => SUCCESS
            [messageKey] => successfullyJoined
            [message] => You have joined successfully.
            [meeting_id] => fe17f66b402b8c178785ebc96fe2fb6e96b17b7a-1543763121613
            [user_id] => w_gxprjs9w5sdg
            [auth_token] => nrmw2bkmxxgo
            [session_token] => ljbp2qalayrpgs3p
        )

)
```
When i want to open manualy
https://server/bigbluebutton/api/enter?sessionToken=ljbp2qalayrpgs3p I got
```
{"response":{"returncode":"FAILED","message":"Could not find conference.","logoutURL":"https://server"}}
```
BUT WHEN i copy - paste $url to browser I see response like $joinMeetingResponse:
```
<response>
<returncode>SUCCESS</returncode>
<messageKey>successfullyJoined</messageKey>
<message>You have joined successfully.</message>
<meeting_id>
fe17f66b402b8c178785ebc96fe2fb6e96b17b7a-1543763121613
</meeting_id>
<user_id>w_d3oxc7tvmrqs</user_id>
<auth_token>8ekeftauoftf</auth_token>
<session_token>e09c8uqeq3lhsdbk</session_token>
</response>
```
Then i used token **e09c8uqeq3lhsdbk** and it works!
https://server/bigbluebutton/api/enter?sessionToken=e09c8uqeq3lhsdbk
```
{"response":{"returncode":"SUCCESS","fullname":"Михаил","confname":"Конференция \"Нейросети\"","meetingID":"fe17f66b402b8c178785ebc96fe2fb6e96b17b7a-1543763457759","externMeetingID":"356a192b7913b04c54574d18c28d46e6395428ab_8f9329121a2844579d346a7f5660372114b08ccf","externUserID":"w_6cepjew2ta42","internalUserID":"w_6cepjew2ta42","authToken":"byinyph7uahw","role":"MODERATOR","guest":false,"conference":"fe17f66b402b8c178785ebc96fe2fb6e96b17b7a-1543763457759","room":"fe17f66b402b8c178785ebc96fe2fb6e96b17b7a-1543763457759","voicebridge":"05635","dialnumber":"613-555-1234","webvoiceconf":"05635","mode":"LIVE","record":"false","isBreakout":false,"logoutTimer":0,"allowStartStopRecording":false,"welcome":"<br>Welcome to <b>Конференция \"Нейросети\"<\u002fb>!<br><br>For help on using BigBlueButton see these (short) <a href=\"event:http://www.bigbluebutton.org/content/videos\"><u>tutorial videos<\u002fu><\u002fa>.<br><br>To join the audio bridge click the phone button.  Use a headset to avoid causing background noise for others.<br><br><br>This server is running <a href=\"http://docs.bigbluebutton.org/\" target=\"_blank\"><u>BigBlueButton<\u002fu><\u002fa>.","customLogoURL":"","customCopyright":"","muteOnStart":false,"logoutUrl":"http://bbb.loc/conferences.html","defaultLayout":"NOLAYOUT","avatarURL":"https://server/client/avatar.png","customdata":[],"metadata":[{"html5autoswaplayout":"true"},{"html5autosharewebcam":"true"},{"html5hidepresentation":"true"}]}}
```
Where can be problem?

**Comment by GhaziTriki** [comment]
@sazanof Did you read the following doc page? https://github.com/bigbluebutton/bigbluebutton-api-php/wiki/API-Join and http://docs.bigbluebutton.org/dev/api.html#join

**Comment by sazanof** [comment]
> @sazanof Did you read the following doc page? https://github.com/bigbluebutton/bigbluebutton-api-php/wiki/API-Join and http://docs.bigbluebutton.org/dev/api.html#join

Yes, but I dont understand, why did not work session token, which i got, when I use **setRedirect(false)**
I wnt use this token manualy in code, without redirect

"There have been requests if it’s possible to embed the Flash client in a “container”" - it (FALSE) works only with hidden container?

**Comment by Ezanor** [comment]
@sazanof 

Hi, i am trying to do the same thing as you but with the html5 client.
When i set redirect to true i can join the session but i when set

`$joinMeetingParams->setRedirect(false);` 

Then i got the response 
` object(SimpleXMLElement)#8 (7) {
    ["returncode"]=>
    string(7) "SUCCESS"
    ["messageKey"]=>
    string(18) "successfullyJoined"
    ["message"]=>
    string(29) "You have joined successfully."
    ["meeting_id"]=>
    string(54) "9f6292e39147428b22520a43b9498425a984d3d5-1550750589420"
    ["user_id"]=>
    string(14) "w_syc9ljsmota9"
    ["auth_token"]=>
    string(12) "h06xbicqypcn"
    ["session_token"]=>
    string(16) "aufbrctukfgelags"
  }`

and when i use the sessionToken : 

https://server/html5client/join?sessionToken=aufbrctukfgelags (it show a 404 not find)

With `[13:36:19:0020] ERROR: clientLogger: User could not log in HTML5, hit 404`
in the console.

i also tried  
https://server/bigbluebutton/api/enter?sessionToken=aufbrctukfgelags

But i got a  {"response":{"returncode":"FAILED","message":"Could not find conference.","logoutURL":"https://fid-bbb-30.fiducial.fr"}}

Any news on that ?

**Comment by GhaziTriki** [comment]
Thank you @sazanof and @Ezanor for reporting this issue.

This issue is related the required `JSESSIONID` cookie in BigBlueButton 2.0. In 2.2 beta this cookie can be enabling `allowRequestWithoutSession` in `bigbluebutton.properties` in your server.

We are working on fixing very soon.

**Comment by GhaziTriki** [comment]
@sazanof and @Ezanor I have implemented a fix in this commit https://github.com/bigbluebutton/bigbluebutton-api-php/commit/b741c45223ffa432da418da5e26be2953ddb99c7#diff-e4d1ea337abdcccbabdda40526fca995R358

After you call the `join` API with `redirect=false` the `JSESSIONID` will automatically be stored. You need to access to it `$bbb->getJSessionId()` then put as cookie in your request header before doing the redircation. Could you please test the master and tell us if it OK?

**Comment by sazanof** [comment]
Thank you very much! I'll test as soon as possible. 😊

**Comment by GhaziTriki** [comment]
@sazanof The fix is in the `2.0.5` release.

**Comment by sazanof** [comment]
Deep tests....=)
redirect false and sessionToken works only if `allowRequestsWithoutSession` = `true`
in
`/usr/share/bbb-web/WEB-INF/classes/bigbluebutton.properties`

https://github.com/bigbluebutton/bigbluebutton-api-php/issues/64
